src/timeseriesflattener/flattened_dataset.py

```diff
@@ -27,69 +27,105 @@ def add_outcome(
+            fallback=fallback,
+            new_col_name=new_col_name,
+            source_values_colname=source_values_colname,
         )
 
-        if new_col_name is None:
-            new_col_name = values_colname
-
-        self.df[f"{new_col_name}_within_{lookahead_days}_days"] = new_col
-
     def add_predictor(
         self,
         predictor_df: DataFrame,
         lookbehind_days: float,
         resolve_multiple: str,
-        fallback: List[str],
-        outcome_colname: str,
+        fallback: float,
+        source_values_colname: str = "val",
+        new_col_name: str = None,
     ):
         """Adds a predictor-column to the dataset
 
         Args:
             predictor_df (DataFrame): Cols: dw_ek_borger, datotid, value if relevant.
-            lookahead_days (float): How far ahead to look for an outcome in days. If none found, use fallback.
-            resolve_multiple (str): What to do with more than one value within the lookahead.
-                Suggestions: earliest, latest, mean, max, min.
+            lookbehind_days (float): How far behind to look for a predictor value in days. If none found, use fallback.
+            resolve_multiple (Callable): How to handle multiple values within the lookbehind window. Takes a a function that takes a list as an argument and returns a float.
+            fallback (List[str]): What to do if no value within the lookahead.
+            new_col_name (str): Name to use for new col. Automatically generated as '{new_col_name}_within_{lookahead_days}_days'.
+        """
+
+        self._add_col(
+            values_df=predictor_df,
+            direction="behind",
+            interval_days=lookbehind_days,
+            resolve_multiple=resolve_multiple,
+            fallback=fallback,
+            new_col_name=new_col_name,
+            source_values_colname=source_values_colname,
+        )
+
+    def _add_col(
+        self,
+        values_df: DataFrame,
+        direction: str,
+        interval_days: float,
+        resolve_multiple: str,
+        fallback: float,
+        new_col_name: str,
+        source_values_colname: str = "val",
+    ):
+        """Adds a value-column to the dataset
+
+        Args:
+            values_df (DataFrame): Cols: dw_ek_borger, datotid, value.
+            direction (str): Whether to look "ahead" or "behind".
+            interval_days (float): How far to look in direction.
+            resolve_multiple (Callable): How to handle multiple values within the lookbehind window. Takes a a function that takes a list as an argument and returns a float.
             fallback (List[str]): What to do if no value within the lookahead.
-                Suggestions: latest, mean_of_patient, mean_of_population, hardcode (qualified guess)
-            outcome_colname (str): What to name the column
+            new_col_name (str): Name to use for new col. Automatically generated as '{new_col_name}_within_{lookahead_days}_days'.
```

Comment:
`col_name` or `colname`?

Reply:
In the description I think it is fair to write out column ;)

Reply:
Haha, fair! I've improved consistency.